frontend/src/components/keyboard/index.js

```diff
@@ -1,32 +1,88 @@
-import React, {useState, useRef, useEffect} from 'react'
+import React, {useState, useRef, useEffect, useCallback} from 'react'
 import { observer } from 'mobx-react'
 import { useStore } from '../../store'
 import {default as SimpleKeyboard} from 'react-simple-keyboard'
 import 'react-simple-keyboard/build/css/index.css'
 import { useTheme } from '@mui/material/styles'
 import './style.css'
-import { display, layout, buttonTheme, buttonAttributes, useStyles, syntethicKeysMap } from './keyboardSettings'
+import {
+  display,
+  layout,
+  buttonTheme,
+  buttonAttributes,
+  useStyles,
+  syntethicKeysMap,
+  defaultLayoutName,
+  shiftLayoutName,
+  capslockLayoutName,
+} from './keyboardSettings'
+
+const isLinuxOS = navigator.userAgent.indexOf('Linux') >= 0
 
 const Keyboard = observer(() => {
     const {
       rosStore: { handleKeyboardShortcuts },
       videoPlayerStore: { isFullscreen, toggleFullscreen},
       navigationStore: { activeTab },
     } = useStore()
-    const [layoutName, setLayoutName] = useState('default')
-    const { classes } = useStyles();
-    const keyboardRef = useRef(null);
+    const keyboardRef = useRef(null)
+    const [shiftPress, setShiftPress] = useState({isPressed: false, isVirtual: false, firstClick: false})
+    const [capslockPress, setCapslockPress] = useState({isPressed: false, isVirtual: false, firstClick: false})
+    const theme = useTheme()
+    
+    const onKeyPress = (button, e) => {
```

Comment:
Let's add some basic jsdocs to press/release logic to describe the problem we're trying to solve here. As it seems a bit tricky and might make the further refactoring and improvements hard w/o tests.

Reply:
done